analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/Binary.cs

```diff
@@ -44,6 +46,52 @@ protected override ProgramState LearnBranchingConstraint(ProgramState state, IBi
         return state;
     }
 
+    protected override ProgramState PreProcess(ProgramState state, IBinaryOperationWrapper operation)
+    {
+        if (operation.LeftOperand is { } leftOperand
+            && operation.RightOperand is { } rightOperand
+            && state[leftOperand]?.Constraint<NumberConstraint>() is { } leftNumber
+            && state[rightOperand]?.Constraint<NumberConstraint>() is { } rightNumber
```

Comment:
We don't need null-checks for (most) of the properties coming from Roslyn CFG. Unless they are `null` by design.

CFG would typically contain `ErrorOperation` in case there's some non-sence present. You can add UT with `undefined+8` if you want.
```suggestion
        if (state[operation.LeftOperand]?.Constraint<NumberConstraint>() is { } leftNumber
            && state[operation.RightOperand]?.Constraint<NumberConstraint>() is { } rightNumber
```